packages/amplify-provider-awscloudformation/src/permission-boundary/permission-boundary.ts

```diff
@@ -0,0 +1,124 @@
+import { $TSContext, stateManager, getPermissionBoundaryArn, setPermissionBoundaryArn } from 'amplify-cli-core';
+import { prompt } from 'inquirer';
+import { getIAMClient } from '../aws-utils/aws-iam';
+import { loadConfiguration } from '../configuration-manager';
+
+export const configurePermissionBoundaryForExistingEnv = async (context: $TSContext) => {
+  setPermissionBoundaryArn(await permissionBoundarySupplier(context));
+};
+
+export const configurePermissionBoundaryForInit = async (context: $TSContext) => {
+  const envName = context.exeInfo.localEnvInfo.envName; // the new environment name
+  if (context?.exeInfo?.isNewProject) {
+    // amplify init
+    // on init flow, set the permission boundary if specified in a cmd line arg, but don't prompt for it
+    setPermissionBoundaryArn(
+      await permissionBoundarySupplier(context, { doPrompt: false, envNameSupplier: () => envName }),
+      envName,
+      context.exeInfo.teamProviderInfo,
+    );
+  } else {
+    // amplfiy env add
+    await rolloverPermissionBoundaryToNewEnvironment(context);
+  }
+};
+
+const permissionBoundarySupplierDefaultOptions = {
+  required: false,
+  doPrompt: true,
+  envNameSupplier: (): string => stateManager.getLocalEnvInfo().envName,
+};
+
+/**
+ * Supplies a permission boundary ARN by first checking headless parameters, then falling back to a CLI prompt
+ * @param context CLI context object
+ * @param options Additional options to control the supplier
+ * @returns string, the permission boundary ARN or an empty string
+ */
+const permissionBoundarySupplier = async (
+  context: $TSContext,
+  options?: Partial<typeof permissionBoundarySupplierDefaultOptions>,
+): Promise<string | undefined> => {
+  const { required, doPrompt, envNameSupplier } = { ...permissionBoundarySupplierDefaultOptions, ...options };
+  const headlessPermissionBoundary = context?.input?.options?.['permission-boundary'];
+
+  const validate = context.amplify.inputValidation({
+    operator: 'regex',
+    value: '^(|arn:aws:iam::(\\d{12}|aws):policy/.+)$',
+    onErrorMsg: 'Specify a valid IAM Policy ARN',
+    required: true,
+  });
+
+  if (typeof headlessPermissionBoundary === 'string') {
+    if (validate(headlessPermissionBoundary)) {
+      return headlessPermissionBoundary;
+    } else {
+      context.print.error('The Permission Boundary ARN specified is not a valid IAM Policy ARN');
```

Comment:
We could make it a hard failure. With the current logic, it will follow up with a prompt which gives the customer a chance to correct the ARN. If the --yes flag is specified then it fails with the error on line 62